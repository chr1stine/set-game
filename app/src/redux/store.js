import reducer from './rootReducer';
import { createStore } from 'redux';

const store = createStore(reducer);

store.subscribe(() => {
  let state = store.getState();
  if (state.collectedSet) {
    store.dispatch({ type: 'set' });
  }
});

export default store;
