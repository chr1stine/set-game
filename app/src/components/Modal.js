import React from 'react';
import { useSelector } from 'react-redux';
import { gameRestarted } from '../redux/gameSlice';

const Modal = () => {
  const { timeEnd, timeStart } = useSelector((state) => state.game);
  const { setCount, hintCount } = useSelector((state) => state.board);
  console.log('sdfsdf');
  function computePlayTime(timeStart, timeEnd) {
    let timeSpent = (timeEnd - timeStart) / 1000; // в секундах
    let hh = timeSpent > 3600 ? timeSpent / 3600 : 0;
    timeSpent %= 3600;
    let mm = timeSpent > 60 ? timeSpent / 60 : 0;
    timeSpent %= 60;
    let ss = timeSpent;
    return `${hh != 0 ? `${hh}ч ` : '' + mm != 0 ? `${mm}м ` : '' + `${ss}с`}`;
  }

  return (
    <div id="modal">
      <div className="modal__content">
        <h3>Игра пройдена!</h3>
        <p>{`Счёт: ${setCount}`}</p>
        <p>{`Подсказок запрошено: ${hintCount}`}</p>
        <p>{`Времени пройдено: ${timeEnd != null ? computePlayTime(timeStart, timeEnd) : ''}`}</p>
        <button
          className="btn btn-primary"
          onClick={() => {
            dispatch(gameRestarted());
          }}>
          Ещё раз
        </button>
      </div>
    </div>
  );
};

export default Modal;
