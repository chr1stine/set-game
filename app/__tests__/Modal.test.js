import renderer from 'react-test-renderer';
import Modal from '../src/components/Modal';
import { Provider } from 'react-redux';
import store from '../src/redux/store';

it('Modal renders', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Modal />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
