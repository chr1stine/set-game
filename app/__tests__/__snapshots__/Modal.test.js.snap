// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
<div
  id="modal"
>
  <div
    className="modal__content"
  >
    <h3>
      Игра пройдена!
    </h3>
    <p>
      Счёт: 0
    </p>
    <p>
      Подсказок запрошено: 0
    </p>
    <p>
      Времени пройдено: 
    </p>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Ещё раз
    </button>
  </div>
</div>
`;
